File: python-wrapper/examples/multimodal_input.py
import asyncio
import os
from ferragent import Agent

async def main() -> None:
    agent = Agent.openai("gpt-5.4-mini", os.environ["OPENAI_API_KEY"],
        instructions="Describe visual evidence and uncertainty precisely.")
    response = await agent.run_multimodal([
        {"type": "text", "text": "What is shown in this image?"},
        {"type": "image_url", "url": "https://upload.wikimedia.org/wikipedia/commons/thumb/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg/640px-Gfp-wisconsin-madison-the-nature-boardwalk.jpg"},
    ])
    print(response["content"])

asyncio.run(main())

File: examples/multimodal_input.rs
//! Send text and an image to a vision-capable model.

use ferragent::llm::openai::OpenAiModel;
use ferragent::{Agent, ContentPart, Message};

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    dotenvy::dotenv().ok();
    let model = OpenAiModel::new("gpt-5.4-mini", std::env::var("OPENAI_API_KEY")?);
    let mut agent = Agent::builder(model)
        .instructions("Describe visual evidence precisely and mention uncertainty.")
        .build();

    let input = Message::user_parts(vec![
        ContentPart::text("What is shown in this image?"),
        ContentPart::image_url(
            "https://upload.wikimedia.org/wikipedia/commons/thumb/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg/640px-Gfp-wisconsin-madison-the-nature-boardwalk.jpg",
        ),
    ]);
    let response = agent.run_message(input).await?;
    println!("{}", response.content.unwrap_or_default());
    Ok(())
}
